### ISSUE TITLE
History Dependencies need update 

### ISSUE THREAD

**Issue by GhostMayor** [ReactTraining/react-router#7633]
Since your library uses the history library of version lower than 5, then when simply downloading the history library, an ordinary user will encounter a problem using Router history = {history}, since the switch will not be updated due to different versions of the library

## Version

5.2.0

## Test Case

https://codesandbox.io/s/charming-sound-28duy?file=/src/App.tsx

## Steps to reproduce

version history: 5.0.0

## Expected Behavior

update switch

## Actual Behavior

no update switch after click in link

**Comment by timdorr** [comment]
You shouldn't normally have to install a separate copy of history. When you do, you should get a warning about a mismatched version when installing, as the package states its dependency being 4.x. Regardless, when v6 launches, it will make use of history 5.0.

**Comment by GhostMayor** [comment]
How do you look at exporting the createbrowserHistory method from your library, this would just solve the problem of different versions of history used, in other words, you would not have to download this library separately.

**Comment by timdorr** [comment]
You should use BrowserRouter. You don't need to use history directly.

**Comment by GhostMayor** [comment]
For example, I need to change the route in redux in middleware programmatically according to certain conditions, for this I pay attention to the off guide I need a router element https://reactrouter.com/web/api/Router, as shown in the documentation, but The versions of the libraries differ since the latest version of history 5.xx and react-router 4.xx, respectively, they work in equal contexts. Why can't you supply createBrowserHistory directly from your library to get the version history you are using, that would fix problems with hunt for a working version of history 



```JSX
import { createBrowserHistory, Router} from 'react-router';


export const history = createBrowserHistory();

ReactDOM.render(
  <Router history={history}>
    <App />
  </Router>,
  node
);
```

**Comment by lionskape** [comment]
@timdorr Seems like the react-router v6 update will not come this year. Maybe it is a good reason to update history at the react-router v5?

**Comment by spacejamjim** [comment]
There are longstanding encoding issues with that were fixed with 5.0. Please update this dep @timdorr

**Comment by spacejamjim** [comment]
https://github.com/ReactTraining/history/releases/tag/v5.0.0 There are longstanding encoding issues fixed in 5.0
